### PR TITLE
OCSADV-288: Stress test implemented.

### PR DESCRIPTION
**Pull request by ricardocastrovidal**

### CODE REVIEW

**Comment by swalker2m** [comment]
In order to run this test case, you have to have some service running on localhost at port 9090?  That seems inconvenient for development since to test any change you have to stop the server, rebuild/deploy and start the server again.  Also when we setup the CI system it will want to build and test code as it is committed so I'm not sure how that would work there.

Can the test create an instance of the dhs service itself and tear it down when finished?

**Comment by swalker2m** [comment]
Can we switch this to be built with sbt?  I realize Scala isn't being used here but ideally all the new Java and/or Scala code would be built with the same build system.

File: DhsTranslatorTest/src/test/java/edu/gemini/dhs/translator/test/DhsTranslatorTest.java [hunk 1]
```diff
@@ -25,7 +33,9 @@
 import com.fasterxml.jackson.databind.node.ObjectNode;
 
 public class DhsTranslatorTest {
+    private static final Integer numberOfThreads = 100;
     private final static String BASE_URI = "http://localhost:9090/axis2/services/dhs/images";
+    private static boolean isNotOk = false;
```
Comment (swalker2m) [inline DhsTranslatorTest/src/test/java/edu/gemini/dhs/translator/test/DhsTranslatorTest.java:38]: I'm not sure this is guaranteed to be correct in a multi-threaded environment but regardless this flag shouldn't be needed.  You can just let the `Callable` fail with an exception and let that be your guide that something went wrong when you get the result from the `Future`.

File: DhsTranslatorTest/src/test/java/edu/gemini/dhs/translator/test/DhsTranslatorTest.java [hunk 2]
```diff
@@ -246,6 +256,43 @@ public void testSetKeywordWrongType() throws ClientProtocolException,
         assertEquals("Operation did not fail.", "error", status.asText());
     }
 
+    @Test
+    public void testConcurrentCalls() throws InterruptedException,
+            ExecutionException {
+        final ConnectionRequester connectionRequester = new ConnectionRequester();
+
+        Callable<String> task = new Callable<String>() {
```
Comment (swalker2m) [inline DhsTranslatorTest/src/test/java/edu/gemini/dhs/translator/test/DhsTranslatorTest.java]: This is just a test case but still a good habit that we've adopted is to mark everything `final` that can be final.  The absence of the `final` modifier then becomes a warning that something tricky is going to happen.

**Comment by swalker2m** [comment]
This PR is small enough and the title is clear but a few words about what you're trying to accomplish in the description of the PR are always appreciated.

**Comment by cquiroz** [comment]
Agree with @swalker2m on having a description on the PR

**Comment by cquiroz** [comment]
I guess having the test case launch the server could be problematic as it runs on C, and more importantly it needs to talk to the dhs libs, is that right?

File: DhsTranslatorTest/src/test/java/edu/gemini/dhs/translator/test/DhsTranslatorTest.java (hunk 2 above)
Comment (cquiroz) [inline DhsTranslatorTest/src/test/java/edu/gemini/dhs/translator/test/DhsTranslatorTest.java]: Are you using the returned `String` on the checks? Perhaps it should be `Callable<Void>`

**Comment by swalker2m** [comment]
Oh, right.  I'd forgotten that tiny detail.  I suppose this kind of testing in the CI server is sort of out of the question then.

**Comment by tpolecat** [comment]
Just FYI, it's ok to go past 80 columns. I have my editor's margin set at 100 and will go past there if it makes the code more understandable.

**Comment by jluhrs** [comment]
The service can simulate internally all the interactions with the DHS Server, so it is still possible to run the tests without a DHS Server. Of course, then only the HTTP front-end will be tested. Still it can be used for CI. But I would put the automatization of the server start/stop on a new Jira task.

**Comment by ricardocastrovidal** [comment]
Last changes added to stress method test.

**Pull request merged**